keras/losses.py

```diff
@@ -1761,6 +1761,24 @@ def _ragged_tensor_sparse_categorical_crossentropy(y_true,
       sparse_categorical_crossentropy, from_logits=from_logits, axis=axis)
   return _ragged_tensor_apply_loss(fn, y_true, y_pred, y_pred_extra_dim=True)
 
+def _check_ground_truth_label_range(y_true):
+  """Checks whether ground truth label lies inside of the range i.e.,
+  either 0 or 1.
+  
+  Args:
+    y_true: Ground truth values. shape = `[batch_size, d0, .. dN]`.
+
+  Returns:
+    Boolean value `True` if ground truth labels lie inside of the range,
+    `False` otherwise.
+  """
+  flatten_y_true_unpacked = tf.unstack(tf.reshape(y_true, [-1]))
+  for label in flatten_y_true_unpacked:
+    if tf.equal(label, 0) or tf.equal(label, 1):
```

Comment:
`if tf.equal(label, 0) or tf.equal(label, 1):` This line will again not work because here two Tensors are being used as bools (note that tf.equal returns a Tensor), which is not allowed. 
+1 for removing any for loops, this causes a big performance loss.

Reply:
Ran it locally, got the following error as I said,
```
tensorflow.python.framework.errors_impl.OperatorNotAllowedInGraphError: using a `tf.Tensor` as a Python `bool` is not allowed: AutoGraph did convert this function. This might indicate you are trying to use an unsupported feature.
```